qt/FaviconCache.cc

```diff
@@ -20,6 +20,43 @@
 ****
 ***/
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+Q_NETWORK_EXPORT bool qIsEffectiveTLD(QStringView domain);
+#endif
+
+namespace
+{
+
+QString getTopLevelDomain(QUrl const& url)
+{
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+
+    auto const host = url.host();
+    auto const dot = QChar(QLatin1Char('.'));
+
+    for (auto dot_pos = host.indexOf(dot); dot_pos != -1; dot_pos = host.indexOf(dot, dot_pos + 1))
+    {
+        if (qIsEffectiveTLD(QStringView(&host.data()[dot_pos + 1], host.size() - dot_pos - 1)))
```

Comment:
I think this is private API?

This is OK for an interim step but if it's private API we might not want to ship with it?

I have a (now-stale) branch at https://github.com/transmission/transmission/pull/1724 that made progress on using a lightweight third-party library instead of Qt for domain resolution, but it needs to be updated. You're welcome to work on it if you want to, but not required or anything, I'll get back to it sooner or later

Reply:
Mentioned this in c111b7e299e358dcd55f4dc7dce8de8744136e4b commit message. Agreed that something else would be a better long-term solution; for now, it's there in Qt 6.2 (LTS).

Reply:
LTS isn't any guarantee that private API goes unchanged, but a longer-term fix is on our radar anyway so IMO this isn't a blocker